spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayMigrationStrategy.java

```diff
@@ -25,10 +28,18 @@
  * @author Andreas Ahlenstorf
  * @author Phillip Webb
  */
-public class FlywayMigrationStrategy {
+public class FlywayMigrationStrategy implements EnvironmentAware {
 
-	public void migrate(Flyway flyway) {
-		flyway.migrate();
-	}
+    private Environment environment;
```

Comment:
I don't like giving a convenience class full access to the `Environment` when all it needs is a flag. Why not injkect `FlywayProperties`?